pkg/services/ngalert/state/manager.go

```diff
@@ -389,6 +359,10 @@ func (i InstanceStateAndReason) String() string {
 }
 
 func (st *Manager) staleResultsHandler(ctx context.Context, evaluatedAt time.Time, alertRule *ngModels.AlertRule, states map[string]*State, logger log.Logger) []*State {
+	// If we are removing two or more stale series it makes sense to share the resolved image as the alert rule is the same.
+	// TODO: We will need to change this when we support images without screenshots as each series will have a different image
+	var resolvedImage *ngModels.Image
```

Comment:
I wonder if the image should be passed as an argument because during this cycle manager could process normal states and already take a screenshot. Another idea is to do this on the screenshot service side (i.e. caching rule + eval timestamp)

Reply:
Another alternative: this method accepts a map of states which may already have the image taken. Probably reuse it?

Reply:
I would not be against passing it as an argument, but there are two issues in doing so that I would need to figure out:

1. Does `ProcessEvalResults` then take the screenshot? We want a screenshot of the graph from around the time the alert was resolved, rather than the original screenshot from when it fired
2. In future when we replace screenshots with server-side images we will need to create an image per state as each state will have different historical time series

Reply:
1 Isn't the image we create in `setNextState` the same as we create in the stale results handler? I think that currently, we can make it once per `ProcessEvalResults` and then share it between all states. Is that correct?
2. Agree but we can refactor when we are there, right?

Reply:
I think it would be an issue when the alert rule is deleted because there would be no next state, however the results handler would still need to resolve all existing alerts.